src/dotnet/commands/dotnet-new3/New3CommandShim.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Config;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.DotNet.Tools.New3
+{
+    internal class New3CommandShim
+    {
+        private const string HostIdentifier = "dotnetcli";
+        private const string HostVersion = "1.0.0";
+        private const string CommandName = "new3";
+
+        public static int Run(string[] args)
+        {
+            return New3Command.Run(CommandName, CreateHost(), FirstRun, args);
+        }
+
+        private static ITemplateEngineHost CreateHost()
+        {
+            var builtIns = new Dictionary<Guid, Func<Type>>
+            {
+                { new Guid("0C434DF7-E2CB-4DEE-B216-D7C58C8EB4B3"), () => typeof(RunnableProjectGenerator) },
```

Comment:
There are two real options here, either we leave the GUIDs as is or we change to bootstrapping the engine from more nupkgs in the layout. The guids are needed to bootstrap an environment that does not support online acquisition of the generators and/or include them in the layout (uses package references instead). Which would we prefer going with? (It sounds like we're leaning towards adding them to the layout to get rid of the GUID mess)